pkg/scheduler/core/generic_scheduler_test.go

```diff
@@ -217,7 +217,7 @@ func TestSelectHost(t *testing.T) {
 }
 
 func TestGenericScheduler(t *testing.T) {
-	algorithmpredicates.SetPredicatesOrdering(order)
+	defer algorithmpredicates.SetPredicatesOrderingDuringTest(order)()
```

Comment:
nit: this is clever, but a bit unreadable. One might think that the whole line is `defer`ed and executed after the function returns. Optionally, you can store and `defer` the returned function, but it is up to you.

Reply:
Credit goes to @liggitt :)

```go
defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.XYZ, true)()
```